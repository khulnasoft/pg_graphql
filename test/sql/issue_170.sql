begin;
    create table account(
        id int primary key
    );

    insert into public.account(id)
    select * from generate_series(1,5);

    -- hasPreviousPage is true when `after` is first element of collection
    -- "WzFd" is id=1
    -- because result set does not include the record id = 1

    select jsonb_pretty(
        graphql.resolve($$
            {
              accountCollection(first: 2, after: "WzFd") {
                pageInfo{
                  hasPreviousPage
                }
              }
            }
        $$)
    );

    -- hasPreviousPage is false when `after` is before the first element of collection
    -- "WzFd" is id=0
    select jsonb_pretty(
        graphql.resolve($$
            {
              accountCollection(first: 2, after: "WzBd") {
                pageInfo{
                  hasPreviousPage
                }
              }
            }
        $$)
    );
rollback;
