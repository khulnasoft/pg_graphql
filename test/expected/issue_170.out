begin;
    create table account(
        id int primary key
    );
    insert into public.account(id)
    select * from generate_series(1,5);
    -- hasPreviousPage is true when `after` is first element of collection
    -- "WzFd" is id=1
    -- because result set does not include the record id = 1
    select jsonb_pretty(
        graphql.resolve($$
            {
              accountCollection(first: 2, after: "WzFd") {
                pageInfo{
                  hasPreviousPage
                }
              }
            }
        $$)
    );
              jsonb_pretty               
-----------------------------------------
 {                                      +
     "data": {                          +
         "accountCollection": {         +
             "pageInfo": {              +
                 "hasPreviousPage": true+
             }                          +
         }                              +
     }                                  +
 }
(1 row)

    -- hasPreviousPage is false when `after` is before the first element of collection
    -- "WzFd" is id=0
    select jsonb_pretty(
        graphql.resolve($$
            {
              accountCollection(first: 2, after: "WzBd") {
                pageInfo{
                  hasPreviousPage
                }
              }
            }
        $$)
    );
               jsonb_pretty               
------------------------------------------
 {                                       +
     "data": {                           +
         "accountCollection": {          +
             "pageInfo": {               +
                 "hasPreviousPage": false+
             }                           +
         }                               +
     }                                   +
 }
(1 row)

rollback;
